events/emitter.go

```diff
@@ -40,19 +41,25 @@ func (e *Emitter) Post(ctx context.Context, evt Event) error {
 
 func (e *Emitter) Events(ctx context.Context, clientID string) chan *events.Envelope {
 	e.m.Lock()
-	if _, ok := e.sinks[clientID]; !ok {
-		ns, _ := namespaces.Namespace(ctx)
-		s := &eventSink{
-			ch: make(chan *events.Envelope),
-			ns: ns,
-		}
-		e.sinks[clientID] = s
-		e.broadcaster.Add(s)
+	defer e.m.Unlock()
+
+	if v, ok := e.sinks[clientID]; !ok {
+		e.broadcaster.Remove(v)
 	}
-	ch := e.sinks[clientID].ch
-	e.m.Unlock()
 
-	return ch
+	ns, _ := namespaces.Namespace(ctx)
+	s := &eventSink{
+		id: clientID,
+		ch: make(chan *events.Envelope),
+		ns: ns,
+	}
+	q := goevents.NewQueue(s)
+	r := goevents.NewRetryingSink(q, goevents.NewBreaker(5, time.Millisecond*100))
```

Comment:
Does a retrying sink make sense in this case? It's wrapping a `Queue`, and writes to a `Queue` can only fail when the queue is closed, in which case retrying doesn't help.